ETL_pipeline/load_flood.py

```diff
@@ -0,0 +1,146 @@
+        location = county
+    if not county:
+        county = location
+    return location, county, country
+
+
+def insert_location(conn: connection, latitude: float, longitude: float) -> int | None:
+    """Insert a location into the database, and it's associated county 
+    and country where they don't already exist."""
+    location, county, country = get_location_names(latitude, longitude)
+    if country:
+        with conn.cursor() as cur:
+            cur.execute(f"""SELECT country_id FROM country
+                            WHERE name = '{country}'""")
+            country_id = cur.fetchone()[0]
+            cur.execute(f"""SELECT county_id FROM county
+                            WHERE name = '{county}'
+                            AND country_id = {country_id}""")
+            county_id = cur.fetchone()
+            if not county_id:
+                cur.execute(f"""INSERT INTO county (name, country_id)
+                                VALUES ('{county}', {country_id})
+                                RETURNING county_id""")
+                conn.commit()
+                county_id = cur.fetchone()
+            county_id = county_id[0]
+            cur.execute(f"""INSERT INTO location (latitude, longitude, loc_name, county_id)
+                            VALUES ({latitude}, {longitude}, '{location}', {county_id})
+                            RETURNING loc_id""")
+            conn.commit()
+            loc_id = cur.fetchone()
+        return loc_id
+    return None
+
+
+def get_location_id(conn: connection, latitude: float, longitude: float) -> int | None:
+    """Obtain the location id from the database."""
+    with conn.cursor() as cur:
+        cur.execute(f"""SELECT loc_id FROM location
+                    WHERE latitude = {latitude}
+                    AND longitude = {longitude}""")
+        loc_id = cur.fetchone()
+    if not loc_id:
+        loc_id = insert_location(conn, latitude, longitude)
+    if loc_id:
+        return loc_id[0]
+    return None
+
+
+def insert_flood(conn: connection, flood: dict) -> None:
+    """Insert a flood into the database."""
+    loc_id = get_location_id(conn, flood['latitude'], flood['longitude'])
+    if loc_id:
+        severity_level_id = flood['severity_level_id']
+        time_raised = flood['time_raised']
+        with conn.cursor() as cur:
+            cur.execute(f"""SELECT * FROM flood_warnings
+                        WHERE severity_level_id = {severity_level_id}
+                        AND time_raised = '{time_raised}'
+                        AND loc_id = {loc_id}""")
+            flood = cur.fetchone()
+            if not flood:
+                cur.execute(f"""INSERT INTO flood_warnings
+                            (severity_level_id, time_raised, loc_id, notified)
+                            VALUES ({severity_level_id}, '{time_raised}', {loc_id}, False)""")
+                conn.commit()
+
+
+def insert_all_floods(config, floods: list[dict]) -> None:
```

Comment:
`config` type hint?

Reply:
what IS the typehint for config/environ?

Reply:
os._Environ ?

Reply:
ok i did dict